src/utils.cpp

```diff
@@ -332,8 +332,153 @@ void cnf_to_dimacs(Term cnf, std::ostringstream & y)
      y << 0;  // Symbolizing end of line
      y << "\n";
   }
+} 
+
+Term to_cnf(Term formula, SmtSolver s)
+{
+  
+  std::map<Term, Term>ma;//map ma contains the mapping of each term to it's new symbol in tseytin's transformation
+  
+  Sort boolsort = formula->get_sort();
+  assert(boolsort->get_sort_kind() == BOOL);
```

Comment:
A minor point here -- this won't always pass due to an edge case for some solvers. Some solvers alias booleans and bit-vectors of size one. And in smt-switch we (somewhat arbitrarily) picked to treat them as bit-vectors. In that case, even if it is a formula, the sort kind would actually be `BV` with width one. Using the logging solver handles this edge case (see [here](https://github.com/makaimann/smt-switch#logging-solvers) for more info), but since we don't know that we'll get a logging solver, I'd recommend the following:
```
Sort boolsort = s->make_sort(BOOL);
assert(formula->get_sort() == boolsort);
```
which should do what you want even if the underlying solver treats booleans and bit-vectors the same.

Reply:
Changed it in the last commit